src/dbnode/persist/fs/seek_manager.go

```diff
@@ -315,6 +352,22 @@ func (m *seekerManager) Borrow(shard uint32, start time.Time) (ConcurrentDataFil
 	return availableSeeker.seeker, nil
 }
 
+func (m *seekerManager) isShardAvailable(shard uint32) bool {
+	m.RLock()
+	defer m.RUnlock()
+	return m.isShardAvailableWithLock(shard)
+}
+
+func (m *seekerManager) isShardAvailableWithLock(shard uint32) bool {
+	shardState, err := m.shardSet.LookupStateByID(shard)
+	// NB(bodu): LookupStateByID returns ErrInvalidShardID when shard
+	// does not exist in the shard map which means the shard is not available.
+	if err != nil {
+		return false
+	}
+	return shardState == shardTypes.Available
```

Comment:
Also I don't think we have to check that state is available. Leaving shards are meant to respond with data.